sympy/combinatorics/coset_table.py

```diff
@@ -1014,3 +1016,169 @@ def coset_enumeration_c(fp_grp, Y, max_cosets=None, draft=None,
+                modified_coincidence(C, f, alpha, f_p**-1*y)
+            return
+        while j >= i and C.table[b][C.A_dict_inv[w[j]]] is not None:
+            b_p = C.P[b][C.A_dict_inv[w[j]]]
+            b = C.table[b][C.A_dict_inv[w[j]]]
+            j -= 1
+        if j < i:
+            modified_coincidence(C, f, b, f_p**-1*b_p)
+        elif j == i:
+            # Dedcution
+            C.table[f][C.A_dict[w[i]]] = b
+            C.table[b][c.A_dict_inv[w[i]]] = f
+            C.P[f][C.A_dict[w[i]]] = f_p**-1*b_p
+            C.P[f][C.A_dict_inv[w[i]]] = b_p**-1*f_p
+        else:
+            if not fill:
+                modified_define(C, f, w[i])
+
+        # For scan an not filling.
+        if not fill:
+            break
+
+def modifed_scan_and_fill(C, alpha, w, y):
+    modified_scan(C, alpha, w, y, fill=True)
+
+def modified_rep(C, k, p_p):
+    '''
+     Input: C, κ∈Ω, , pP
+    '''
+    p = C.p
+    lambda_ = k
+    rho = p[lambda_]
+    # Array to trace back compression path.
+    s = p[:]
+    while rho != lambda_:
+        s[rho] = lambda_
+        lambda_ = rho
+        rho = p[lambda_]
+    rho = s[lamda]
+    while rho != k:
+        mu = rho
+        rho = s[mu]
+        p[rho] = lambda_
+        p_p[rho] = p_p[rho]*p_p[mu]
+    return lambda_
+
+def modified_merge(C, k, lamda, w, p_p, q, l):
+    '''
+    Input:κ,λ∈Ω,w∈(Y∪Y–1)*withτ(κ)= φ(w)τ(λ),p_p,q,l GP
+    q is a queue of length l of elements to be deleted from Ω *
+    '''
+    phi = modified_rep(C, k, p_p)
+    psi = modified_rep(C, lamda, p_p)
+    if phi > psi:
+        C.p[phi] = psi
+        p_p[phi] =  p_p[k]**-1*w*p_p[lamda]
+        l += 1
+        q[l] = psi
+    elif phi < psi:
+        C.p[psi] = phi
+        p_p[psi] = p_p[lamda]**-1*w**-1*p_p[k]
+        l += 1
+        q[l] = psi
+
+def modified_coincidence(C, alpha, beta, w):
+    '''
+    Input: A coincident pair α, ß∈Ω, w∈ (Y∪Y–1)*
+    '''
+    l = 0
+    i = 0
+    q = []
+    p_p = []
+    table = C.table
+    modified_merge(C, alpha, beta, w, p_p, q, l)
+    while len(q):
+        gamma = q.pop(0)
+        for x in C.A_dict:
+            delta = C.table[gamma][C.A_dict[x]]
+        if delta:
+            table[delta][C.A_dict_inv[x]] = None
+            mu = modified_rep(C, gamma, p_p)
+            nu = modified_rep(C, delta, p_p)
+        if table[mu][C.A_dict[x]]:
+            v = p_p[delta]**-1*C.P[gamma][C.A_dict[x]]**-1*p_p[gamma]*C.P[mu][C.A_dict[x]]
+            modified_merge(C, nu, table[mu][C.A_dict[x]], v, p_p, q, l)
+        elif table[nu][C.A_dict_inv[x]]:
+            v = p_p[gamma]**-1*C.P[gamma][C.A_dict[x]]*p_p[delta]*C.P[mu][C.A_dict_inv[x]]
+            modified_merge(C, mu, table[nu][C.A_dict_inv[x]], v, p_p, q, l)
+        else:
+            table[mu][C.A_dict[x]] = nu
+            table[nu][C.A_dict_inv[x]] = mu
+            v = p_p[gamma]**-1*C.P[gamma][C.A_dict[x]]*p_p[delta]
+            C.P[mu][C.A_dict[x]] = v
+            C.P[nu][C.A_dict_inv[x]] = v**-1
```

Comment:
All of the above functions should be methods of `CosetTable`, for consistency.

Reply:
Okay 👍

Reply:
These are still not methods of `CosetTable` even though you pushes a couple of commits since.

Reply:
I've updated that now.